### Long read assembly

Using long read sequencing data such as PacBio or nanopore in combination with high coverage illumina shot-gun sequencing for correction yields useful genome assemblies. For the first assembly, I use `Canu`, which allows to input sequencing data from different sources (PacBio, nanopore, BAC clones). 

```ShellSession
#on genotoul, run these lines first
module load bioinfo/Java8;
module load compiler/gcc-6.1.0;

#the script I used for nanopore (on genotoul)
/usr/local/bioinfo/src/canu/canu-1.6/Linux-amd64/bin/canu -java=/tools/java/jre1.8.0_45/bin/java -p ./prefix_of_genome -d /PATH-to-genome-folder/ genomeSize=40m gridEngineThreadsOption='-pe parallel_smp THREADS' maxMemory=64G maxThreads=8 -nanopore-raw nanopore-run.fastq
```

Then, the assembly should be polished using Illumina shot-gun sequencing data. First, the reads should be mapped to the new assembly and a sorted bam-file created with `samtools`. For polishing, I use `Pilon`.
```ShellSession
java -Xmx8G -jar /usr/local/bioinfo/src/pilon/pilon-1.22.jar --genome canu_contigs.fasta --frags illumina-reads.bam --output canu_contigs.pilon.fa
```

Three rounds of polishing (and re-mapping of the Illumina reads to the polished versions) are usually sufficient, more rounds do not significantly improve the assembly any more. 


### Short read assembly

If only Illumina short-read sequencing data is available, a short-read assembler is required. I used `SPAdes` for this:

```ShellSession
spades.py -1 read_1_trimmed.fq -2 read_2_trimmed.fq -s reads_unpaired_trimmed.fq --careful -o output.spades
```
