## Long read assembly
#### Assembly with `Canu`
Using long read sequencing data such as PacBio or nanopore in combination with high coverage illumina shot-gun sequencing for correction yields useful genome assemblies. For the first assembly, I use `Canu`, which allows to input sequencing data from different sources (PacBio, nanopore, BAC clones). 

```ShellSession
export HOME=/home/sk893857 #setting up home directory
canu -java=/tools/java/jre1.8.0_45/bin/java -p ./MySu_canu-v0 -d $HOME/genomes/Msulcatula genomeSize=40m gridEngineThreadsOption='--cpus-per-task=THREADS' maxMemory=120G maxThreads=8 -nanopore-raw 20171213_mysul-G10-S00.fastq.gz
```
#### Polish the assembly with `Pilon`
Then, the assembly should be polished using Illumina shot-gun sequencing data. First, the reads should be mapped to the new assembly and a sorted bam-file created with `samtools`. For polishing, I use `Pilon`.
```ShellSession
java -Xmx8G -jar $HOME/miniconda3/bin/pilon-1.22.jar --genome MySu_canu-v0_contigs.fasta --frags illumina-reads.bam --output canu_contigs.pilon.fa
```

In case of *Myriosclerotinia sulcatula*, I performed three rounds of polishing (including re-mapping of the Illumina reads to the polished versions), more rounds did not significantly improve the assembly any more. 


## Short read assembly
#### Assembly with `SPAdes`
If only Illumina short-read sequencing data is available, a short-read assembler is required. I used `SPAdes` for this:

```ShellSession
spades.py -1 read_1_trimmed.fq -2 read_2_trimmed.fq -s reads_unpaired_trimmed.fq --careful -o output.spades
```

#### Assembly with `SOAPdenovo2`
```ShellSession

```
