No assembly is perfect. To assess the quality of a genome assembly, several values can be assessed:

- Genome size (close to expectation?)
- number of contigs
- N50/N90 as measure of median contig size
- coverage
- completeness

#### Genome statistics
For a quick overview over some genome statistics, I used `assembly-stats`:
```ShellSession
assembly-stats Mysul_MySu01_v1.fa
```
The output includes N50, N90, and other size stats: 
```
sum = 43533135, n = 24, ave = 1813880.62, largest = 5732858
N50 = 2649720, n = 6
N60 = 2389439, n = 8
N70 = 2109168, n = 10
N80 = 1913538, n = 12
N90 = 1133061, n = 15
N100 = 13241, n = 24
N_count = 0
Gaps = 0
```

#### Re-mapping long reads
It can be useful to re-map long reads from PacBio or nanopore to assess the coverage of contigs. `bwa` is a tool that allows this:
```ShellSession
bwa index Mysul_MySu01_v1.fa
bwa mem -x ont2d Mysul_MySu01_v1.fa 20171213_mysul-G10-S00.fastq.gz > align-nanopore.sam
samtools sort -l 0 -o align-nanopore.bam -O bam align-nanopore.sam #create the BAM file
samtools view -F 0x04 -b align-nanopore.bam > align-nanopore.aligned.bam #clean up BAM file from non-mapping reads
```

#### BLASTN against NCBI nt database
Using `NCBI BLAST+ 2.7` and the latest version of the NCBI nucleotide collection database nt, I aligned the contigs against the known nucleotide sequences. This can help identify contigs of other origin (contaminant, viruses, mitochondrial genome). 
```ShellSession
blastn -task megablast -query Mysul_MySu01_v1.fa -db $HOME/blastdb/nt -outfmt '6 qseqid staxids bitscore std sscinames sskingdoms stitle' -culling_limit 5 -num_threads 32 -evalue 1e-25 -out MySu01_v1.blast-nt.tbl
```

#### Visualizing data with `Blobtools`
`Blobtools` is a very useful tool to summarize and visualize BLAST-nt and coverage data, and helps rapidly identify contaminant contigs. The putative identity of contigs can be generated on the levels of superkingdom, class, phylum, family, and genus.
```ShellSession
blobtools create -i Mysul_MySu01_v1.fa -b align-nanopore.aligned.bam -t MySu01_v1.blast-nt.tbl -o blobfile --db $HOME/miniconda3/opt/blobtools-1.0.1/data/nodesDB.txt #creates the JSON file
blobtools blobplot -i blobfile.blobDB.json -o MySu01_plot -r family --format pdf #to visualize as PDF
blobtools view -i blobfile.blobDB.json -r family -o MySu01_table #the table helps to inspect statistics (coverage, identity, GC content) of contigs
```

The plot looks like this:
![blobplot_mysul](https://github.com/stefankusch/Genome.assembly-Myriosclerotinia/blob/master/Mysul_MySu01_v1_blobtools_family.Mysul_MySu01_v1_blobtools.blobDB.json.bestsum.family.p7.span.100.blobplot.bam0.pdf)

#### Genome coverage

The coverage can be measured with `bedtools` and a bam-file where genome reads were re-mapped to the assembly. Another file contains contig or chromosome sizes, using the same contig/chromosome names as the bam-file. 

```ShellSession
genomeCoverageBed -ibam genome_remapping.bam -g chr-sizes.txt > coverage.txt
```


#### Assessing completeness with CEGMA and BUSCO

Both CEGMA and BUSCO estimate completeness by searching for the presence and completeness of highly conserved gene sets. CEGMA tests the genome fasta file against a protein database.
```ShellSession
cegma --genome new.genome.fa -o out.file -threads 8
```
BUSCO is a python script. `--out` sets the prefix of the output files, `--in` defines the input (genome fasta file), `--lineage` refers to the reference genome set (here, Ascomycota for conserved fungal genes), `-m` is the input type. 
```ShellSession
# to load the modules (genotoul server)
source /usr/local/bioinfo/src/BUSCO/BUSCO_V3/busco_v3_env/bin/activate;
module load bioinfo/BUSCO_v3;
# to run the script
python /usr/local/bioinfo/src/BUSCO/BUSCO_V3/busco/scripts/run_BUSCO.py -f --out out.file_ --in new.genome.fa --lineage /usr/local/bioinfo/src/BUSCO/datasets/ascomycota_odb9/ -m genome
```
