Proper gene annotation of a newly sequenced genome requires masking of repetitive elements (transposons), then automated gene prediction, followed by manual curation. Tools for gene prediction include `augustus`, `MAKER2`, and `BRAKER1`. 

### Masking repeats with `repeatmasker`

This step is important to label transposable elements that might mess up the annotation, and additionally can provide information about the nature and distribution of different types of transposons. In the following line, the analysis was done for a fungal species. `RepeatMasker -h` provides a list of all functions. 
```ShellSession
RepeatMasker -species Fungi -small -nolow -gff -dir PATH-for-RepeatMasker-directory/ PATH/genome.fa
```
`-nolow` prevents masking of low complexity and simple repeats, and is optional. `-gff` generates an additional GFF output file, which is important for the manual annotation. `-small` returns masked sequence in lower case (standard is N, other options are available). The output will be a masked.fa fasta file meeting the settings, plus the GFF file, a .out file giving localization and statistics of each repeat, and a .tab file summarizing the results of the masking. 


### Gene prediction with `augustus`

A pretty extensive tutorial on the usage of `augustus` can be found here: http://bioinf.uni-greifswald.de/augustus/binaries/tutorial/. Most `perl`-scripts are provided in the scripts/ folder of `augustus`. 

   - **training `augustus` with gene models**

Assuming that good gene annotation of a close relative is present, we can start training `augustus`. First, I generated a GeneBank file from the GFF containing the gene annotations, and produce a random training set of 100 genes, `ref_species.genes.gb.test`, and a training set `ref_species.genes.gb.train` excluding these genes:
```ShellSession
perl PATH/gff2gbSmallDNA.pl ref_species.annotation.gff ref_species.genes.gb;
perl PATH/randomSplit.pl ref_species.genes.gb 100
```
Then, I generated a species file `ref_species`:
```ShellSession
perl PATH/new_species.pl --species=ref_species
```
The output file `ref_species_parameters.cfg` can be set to include protein and codingseq information, hints from RNA-Seq and UTR prediction. 
Next, an initial round of training can be done.
```ShellSession
etraining --species=ref_species ref_species.genes.gb.train
```
This updates the species file. Next, a first test is done with the gb.test file from above:
```ShellSession
augustus --species=ref_species ref_species.genes.gb.test | tee firsttest.out
```
The result of this test can be read with the command below, where TP shows the number of correctly predicted genes, FN false negaties, FP false positives, sensitivity the percentage of correctly predicted exons, and specificity the percentage of exons exactly as in test set.
```ShellSession
grep -A 22 Evaluation firsttest.out
```
Next, we run the following lines to optimize prediction accuracy of `augustus`, and then re-train the program with `etraining` followed by testing the accuracy as above:
```ShellSession
optimize_augustus.pl --species=ref_species ref_species.genes.gb.train #may take a long time
```
This should be done several times until the accuracy does not improve significantly any more. 


   - ***ab initio* gene prediction**

Using the training database created above, `augustus` can be used to predict gene models *ab initio*. 
```ShellSession
augustus --species=ref_species new_genome.masked.fa --extrinsicCfgFile=extrinsic.ref_species.cfg > augustus.abinitio.gff
```
This might take a moment to compute. The output is a .gff file containing information about putative gene models. The protein sequences can be extracted with 
```ShellSession
perl PATH/augustus/augustus-3.2.3/scripts/getAnnoFasta.pl augustus.abinitio.gff
```
producing a file `augustus.abinitio.aa` in fasta format. To extract gene and CDS sequences, I used `gff2fasta.pl` from https://github.com/ISUgenomics/common_scripts/blob/master/gff2fasta.pl (save script with an editor like nano).
```ShellSession
perl PATH/gff2fasta.pl new_genome.fa augustus.abinitio.gff output.prefix
```

   - **including RNA-Seq for prediction**

This requires a BAM and a WIG file(s) produced by mapping of your RNA-Seq against the new genome with a mapper like `STAR` or `HISAT2`. First, RNA-Seq hints have to be produced. There are many options specified in the manual that should be considered for optimization.
```ShellSession
cat PATH/rnaseq.wig | perl PATH/augustus/augustus-3.2.1/scripts/wig2hints.pl --width=10 --margin=10 --minthresh=2 --minscore=4 --src=W --type=ep --radius=4.5 > rnaseq.track.gff
```
This script generates the file `extrinsic.new_species.cfg`, which can be adjusted (see manual for details!). For example, in the line exonpart, the last number can be set as bonus (>1) if the gene model is supported by the RNA-Seq data.
```Text
exonpart        1     .997  M    1  1e+100  E 1    1  W 1    1.05
```
Then, we can predict gene models:
```ShellSession
augustus --species=ref_species new_genome.masked.fa --extrinsicCfgFile=extrinsic.new_species.cfg --hintsfile=hints.rnaseq.ep.gff > augustus.hints.abinitio.gff
```
The gene models can be extracted from the .gff as above. 


### Gene prediction with `BRAKER1`

   - ***ab initio* gene prediction**

`BRAKER` requires both a reference gene set and RNA-Seq data, and you need to refer to `Genemark`, `bamtools` and `samtools` binaries:
```ShellSession
perl PATH/BRAKER/BRAKER_v1.9/braker.pl --cores=8 --fungus --GENEMARK_PATH=PATH/bin/gm_et_linux_64/gmes_petap --BAMTOOLS_PATH=PATH/bamtools/pezmaster31-bamtools-6708a21/bin --SAMTOOLS_PATH=PATH/samtools/samtools-1.3.2 --overwrite --useexisting --species=ref_species --genome=PATH/new_genome.masked.fa --bam=PATH/sortedByCoord.out.bam
```
