This document is meant to summarize what input material I used and how I parsed the files to upload to an instance of `WebApollo`. 

## Input data
#### *Ab initio* gene predictions
`BRAKER` and `Augustus` produce gff annotation files of the predicted gene models. I have used the GeneMark and Augustus output produced by `BRAKER`, which needs parsing to gff3 as follows: 
```ShellSession
gffread -E MySu01_BRAKER-Augustus.gtf -o- > MySu01_BRAKER-Augustus.gff3
gffread -E MySu01_BRAKER-genemark.gtf -o- > MySu01_BRAKER-genemark.gff3
```
The gff3 file should look like this: 
```
# gffread -E genemark.gtf -o-
# gffread v0.10.1
##gff-version 3
tig00000001	GeneMark.hmm	mRNA	21	865	.	+	.	ID=1_t;geneID=1_g
tig00000001	GeneMark.hmm	exon	21	50	.	+	.	Parent=1_t
tig00000001	GeneMark.hmm	exon	147	865	.	+	.	Parent=1_t
tig00000001	GeneMark.hmm	CDS	21	50	.	+	2	Parent=1_t
tig00000001	GeneMark.hmm	CDS	147	865	.	+	2	Parent=1_t
```

#### Repeat landscape
text

#### BLAST similarities
text

#### RNA-seq coverage
text

#### RNA-seq reads
text

## Initializing genome in `WebApollo`
#### Setting up the genome
First, a directory for the new genome has to be set-up with `mkdir ~/opt/Apollo-2.0.6/data_mysul`. Then, for uploading the genome:
```ShellSession
bin/prepare-refseqs.pl --fasta $HOME/data/Mysul_MySu01_v1.fa --out ~/opt/Apollo-2.0.6/data_mysul/
```

## Uploading data
#### Annotations (Repeats, gene predictions, BLAST)
In the Apollo directory, the annotations can be uploaded with the supplied scripts (`WebApollo-2.0.6`). The gene annotations were uploaded like this:
```ShellSession
bin/flatfile-to-json.pl -gff $HOME/data/MySu01_BRAKER-Augustus.gff3 --type mRNA -tracklabel BRAKER-Augustus --out ~/opt/Apollo-2.0.6/data_mysul/
bin/flatfile-to-json.pl -gff $HOME/data/MySu01_BRAKER-genemark.gff3 --type mRNA -tracklabel BRAKER-Genemark --out ~/opt/Apollo-2.0.6/data_mysul/
```
The Repeat landscape from `RepeatMasker` was uploaded as follows:
```ShellSession
bin/flatfile-to-json.pl -gff $HOME/data/MySu01_v1.RepeatMasker.gff --type similarity -tracklabel RepeatMasker --out ~/opt/Apollo-2.0.6/data_mysul/
```
BLAST results were added:
```ShellSession
bin/flatfile-to-json.pl -gff $HOME/data/MySu01.BLASTN.gff --type match_part -tracklabel BLASTN --out ~/opt/Apollo-2.0.6/data_mysul/
```
