An important part of comparative genome analysis is **synteny**. I used `MUMmer` v3.23 for that until now. 
```ShellSession
#this module is required to run MUMmer on the genotoul
module load compiler/gcc-6.1.0;
#the actual MUMmer script - here nucmer for somewhat similar multi-fasta files
/usr/local/bioinfo/src/MUMmer/current/nucmer -p prefix_output PATH/reference_genome.fasta new_genome.fasta
```

The output will be a `.delta` file. It may be useful to run the script also for each chromosome of the reference, depending how many contigs the new genome has. To make the `.delta` files readable, I use the following lines:
```ShellSession
for infile in *delta
do
  j=$(basename $infile .delta)
  delta-filter -l 1000 -q ${infile} > ${j}_filter.delta
  show-coords -c -l -L 1000 -r -T ${j}_filter.delta > ${j}_filter_coords.txt
  mummerplot -f -p ${j}_filter --postscript ${j}_filter.delta
done
```

This creates a filtered `.delta` file that includes only matches of at least 1000 bp, a text file with coordinates that can be useful for later plotting, and a scatterplot using the `mummerplot` function, written as postscript. 
